lib/hooks/userspace/hook-mongodb-core.js

```diff
@@ -43,21 +48,45 @@ function wrapWithLabel(label) {
         agent.logger.warn('Cannot trace mongo outside of a supported framework.');
         return original.apply(this, arguments);
       }
-      var span = agent.startSpan(label, { db: ns });
+      var labels = { db: ns };
+      if (agent.config_.enhancedDatabaseReporting) {
+        labels.operations = JSON.stringify(ops);
+      }
+      var span = agent.startSpan(label, labels);
       if (typeof options === 'function') {
         return original.call(this, ns, ops,
-          wrapCallback(span, options));
+          wrapCallback(span, options, 127));
```

Comment:
Mumble about hard-coded constant. I would prefer this to be exposed in the config so that the user can customize this if necessary.